src/Uno.UI.TestComparer/Comparer/TestFilesComparer.cs

```diff
@@ -257,74 +248,90 @@ private IEnumerable<string> EnumerateFiles(string path, string pattern)
 			}
 		}
 
-		private void WriteImage(string diffPath, byte[] diff, BitmapFrame frameInfo, int stride)
+		private static IEnumerable<T> LogForeach<T>(IEnumerable<T> q, Action<T> action)
 		{
-			using (var stream = new FileStream(diffPath, FileMode.Create))
+			foreach (var item in q)
 			{
-				var encoder = new PngBitmapEncoder();
-
-				encoder.Interlace = PngInterlaceOption.On;
-
-				var frame = BitmapSource.Create(
-					pixelWidth: (int)frameInfo.PixelWidth,
-					pixelHeight: (int)frameInfo.PixelHeight,
-					dpiX: frameInfo.DpiX,
-					dpiY: frameInfo.DpiY,
-					pixelFormat: frameInfo.Format,
-					palette: frameInfo.Palette,
-					pixels: diff,
-					stride: stride
-				);
-
-				encoder.Frames.Add(BitmapFrame.Create(frame));
-				encoder.Save(stream);
+				action(item);
+				yield return item;
 			}
 		}
 
-		private byte[] DiffImages(byte[] currentImage, byte[] previousImage, int pixelSize)
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static unsafe bool DiffImages(string a, string b, string diffPath)
 		{
-			for (int i = 0; i < currentImage.Length; i++)
+			uint changes = 0;
+			using var img1 = SKBitmap.Decode(@"\\?\" + a);
+			using var img2 = SKBitmap.Decode(@"\\?\" + b);
+
+			var pixmap1 = img1.PeekPixels();
+			// Adjuts color
+			if (img1.ColorType != SKColorType.Rgba8888)
 			{
-				currentImage[i] = (byte)(currentImage[i] ^ previousImage[i]);
+				using (var old = pixmap1)
+					pixmap1 = old.WithColorType(SKColorType.Rgba8888);
 			}
 
-			if (pixelSize == 4)
+			// Adjuts color
+			var pixmap2 = img2.PeekPixels();
+			if (img1.ColorType != SKColorType.Rgba8888)
```

Comment:
Should have been:
```suggestion
			if (img2.ColorType != SKColorType.Rgba8888)
```

But best in a separate method.